src/client/opamAdminCheck.ml

```diff
@@ -93,10 +93,11 @@ let formula_of_pkglist packages = function
     let nvs = List.map OpamCudf.cudf2opam (p::ps) in
     Atom
       (name,
-       OpamFormula.formula_of_version_set
-         (OpamPackage.versions_of_name packages name)
-         (OpamPackage.versions_of_packages
-            (OpamPackage.Set.of_list nvs)))
+       OpamFormula.ors
+         (OpamPackage.Set.fold
+            (fun pkg acc -> OpamFormula.Atom (`Eq, OpamPackage.version pkg) :: acc)
+            (OpamPackage.Set.of_list nvs)
+            []))
```

Comment:
Hm the original reason for this was to avoid extremely long enumerations of all the possible versions of a package. I am wondering why that doesn't seem to be the case here ?
IIRC that would happen in particular on versions of `core`, maybe `ocaml` as well. `core` in particular is bringing multiple layers of dependent packages with synchronised versions.